src/main.rs

```diff
@@ -233,6 +234,55 @@ fn color_for_level(lvl: &JobLogLevel) -> Color {
     }
 }
 
+enum JobBehaviour {
+    CancelLava,
+    LeaveRunning,
+}
+
+struct JobCanceller {
+    behaviour: JobBehaviour,
+    lava: Arc<ThrottledLava>,
+}
+
+impl JobCanceller {
+    pub fn new(
+        lava: Arc<ThrottledLava>,
+        job: &Job,
+        default_behaviour: JobBehaviour,
+    ) -> Result<Self, ()> {
+        let behaviour = if let Some(v) = job.variable("LAVA_CANCELLATION") {
```

Comment:
This kind should really be done in `new_job` where all other env vars are parsed.. For as a future refactoring of the runner i'd like to have it have helpers for command line parsing and env parsing probably building on clap.  So centralizing all this kind of parsing would make it easier in the future to handle.